includes/wc-user-functions.php

```diff
@@ -155,19 +155,60 @@ function wc_create_new_customer_username( $email, $new_user_args, $suffix = '' )
 		$username_parts[] = sanitize_user( $email_username, true );
 	}
 
-	$username = wc_strtolower( implode( '', $username_parts ) );
+	$username = wc_strtolower( implode( '.', $username_parts ) );
```

Comment:
This one is just for precaution, in case of generating awkward usernames by just join all values.